msg.go

```diff
@@ -388,6 +401,9 @@ Loop:
 			if ptr++; ptr > 10 {
 				return "", lenmsg, &Error{err: "too many compression pointers"}
 			}
+			// pointer should guarantee that it advances and points forwards at least
+			// but the condition on previous three lines guarantees that it's
+			// at least loop-free
 			off = (c^0xC0)<<8 | int(c1)
```

Comment:
Probably out of scope for this PR, but [RFC 1035](https://tools.ietf.org/html/rfc1035#section-4.1.4) requires compression pointers to reference _prior_ occurrences, so the arbitrary 10-hop limit could be removed by tracking the starting offset of the most recent pointer and adding a second check at the top of the loop like
```go
if off >= ptrOffset {
	return "", lenmsg, &Error{err: "non-prior compression pointer"}
}
```